### DISCUSSION TITLE
Performance issues with Nominatim

### DISCUSSION THREAD

**Discussion by ghost** [osm-search/Nominatim#2894] [General]
Hello, I had read that in terms of performance NGINX was the best choice for a "high" amount of requests to the server. I am currently aiming to do around 40 requests per second on a 64GB/ 16 core machine using latest Nominatim with world import.

Issue is after 15s the API freezes returning me a 504 Time Out Nginx. I know this isn't NGINX support but was wondering if you had any tips to improve performance for Nominatim using NGINX?

Thank you

**Comment by ghost** [comment]
I think actually it might be the Database having issues, here is my NGINX status:

```
Active connections: 82 
server accepts handled requests
              1077      1077       2634 
Reading: 0 Writing: 81 Waiting: 1 
```

I upgrade the machine to 32C/128GB, and under load the API still takes sometimes 20s to get a response from a lookup. Any ideas how to improve this?

**Comment by otbutz** [comment]
Did you follow the database tuning advices? https://nominatim.org/release-docs/develop/admin/Installation/#tuning-the-postgresql-database

**Comment by ghost** [comment]
Hello, yes here is my current config: https://hastebin.com/semepewexe.ini

**Comment by otbutz** [comment]
SSD/HDD?

**Comment by ghost** [comment]
E2-Standard-32 with SSD persistent disk

**Comment by ghost** [comment]
Is it possible to find the Nominatim config used by OSM somewhere?

**Reply by mtmail** [reply]
https://github.com/openstreetmap/chef/blob/master/roles/nominatim.rb

**Comment by ghost** [comment]
Thanks even with the updated config, my concurrent requests are almost x8 slower, but my system performance are still low. I don't see what I need to increase/unlock..

Current config: https://hastebin.com/qupevokobu.ini

<img width="1083" alt="Capture d’écran 2022-11-24 à 15 34 01" src="https://user-images.githubusercontent.com/92714362/203809216-02badd52-6c51-419f-a882-19c77f8c7fc6.png">


<img width="1087" alt="Capture d’écran 2022-11-24 à 15 32 51" src="https://user-images.githubusercontent.com/92714362/203808997-8f3a26a0-ec52-40fd-afe0-04c9f22afbe1.png">

**Comment by ghost** [comment]
The only thing I didn't modify was the shmmax and shmall because they seem to be higher than the one in the config? Is that good or should I also change it?

```
cat /proc/sys/kernel/shmmax
18446744073692774399
```

```
cat /proc/sys/kernel/shmall
18446744073692774399
```
